### PR TITLE
feat(comp: skeleton): add Skeleton component

### PR DESCRIPTION
**Pull request by brenner8023**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IDuxFE/idux/blob/main/packages/site/src/docs/Contributing.zh.md#commit)
- [x] Tests for the changes have been added/updated or not needed
- [x] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

add Skeleton component

#### SkeletonProps

| 名称 | 说明 | 类型  | 默认值 | 全局配置 | 备注 |
| --- | --- | --- | --- | --- | --- |
| `animated` | 是否开启动画 | `boolean` | `true` | ✅ | - |
| `type` | 骨架的类型 | `'text' \| 'rect' \| 'round' \| 'circle'` | `text` | - | - |
| `loading` | 是否显示加载结束后的UI | `boolean` | `true` | - | - |
| `width` | 骨架的宽度 | `string \| number` | - | - | - |
| `height` | 骨架的高度 | `string \| number` | - | - | - |
| `repeat` | 骨架的重复次数 | `number` | `1` | - | - |

#### SkeletonSlots

| 名称 | 说明 | 参数类型 | 备注 |
| --- | --- | --- | --- |
| default | 展示加载结束后的UI | - | - |

### CODE REVIEW

File: packages/components/skeleton/demo/Loading.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title:
+  zh: 加载结束后切换UI
+order: 1
+---
+
+## zh
+
+支持使用`v-if`或者默认插槽实现切换UI，默认插槽实现会额外生成一个div。
```
Comment (danranVm) [inline packages/components/skeleton/demo/Loading.md:9]: 这里其实不用 额外生成一个 div.

File: packages/components/skeleton/docs/Index.zh.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#### SkeletonProps
+
+| 名称 | 说明 | 类型  | 默认值 | 全局配置 | 备注 |
+| --- | --- | --- | --- | --- | --- |
+| `animated` | 是否开启动画 | `boolean` | `true` | ✅ | - |
+| `type` | 骨架的类型 | `'text' \| 'rect' \| 'round' \| 'circle'` | `text` | - | - |
+| `loading` | 是否显示加载结束后的UI | `boolean` | `true` | - | - |
+| `width` | 骨架的宽度 | `string \| number` | `text` | - | - |
+| `height` | 骨架的高度 | `string \| number` | `text` | - | - |
+| `repeat` | 骨架的重复次数 | `number` | `1` | - | - |
```
Comment (danranVm) [inline packages/components/skeleton/docs/Index.zh.md:30]: 注意 props 的顺序，参见 https://github.com/IDuxFE/idux/issues/225

File: packages/components/skeleton/docs/Index.zh.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+#### SkeletonSlots
+
+| 名称 | 说明 | 参数类型 | 备注 |
+| --- | --- | --- | --- |
+| default | 展示加载结束后的UI | - | - |
+
+#### SkeletonMethods
```
Comment (danranVm) [inline packages/components/skeleton/docs/Index.zh.md:38]: 空的就移除掉。

File: packages/components/skeleton/src/Skeleton.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    const config = useGlobalConfig('skeleton')
+    const loaderClass = computed(() => {
+      const prefixCls = mergedPrefixCls.value
+      return normalizeClass({
+        [`${prefixCls}-loader`]: true,
+        [`${prefixCls}-${props.type}`]: true,
+        [`${prefixCls}-animated`]: props.animated ?? config.animated,
+      })
+    })
+    const style = computed(() => {
```
Comment (danranVm) [inline packages/components/skeleton/src/Skeleton.tsx:31]:
```suggestion
    const loaderStyle = computed(() => {
```

File: packages/components/skeleton/src/Skeleton.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      if (props.type === 'circle') {
+        const size = width || height
+        width = size
+        height = size
+      }
+
+      return {
+        ...(width && { width }),
+        ...(height && { height }),
+      }
```
Comment (danranVm) [inline packages/components/skeleton/src/Skeleton.tsx:43]:
```suggestion
      const { width, height, type } = props
      if (type === 'circle') {
        const size = convertCssPixel(width || height)
        return { width: size, height: size }
      }

      return {
        width: convertCssPixel(width),
        height: convertCssPixel(height),
      }
```

File: packages/components/skeleton/src/Skeleton.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    })
+
+    return () => {
+      const prefixCls = mergedPrefixCls.value
+      const loader = <div class={loaderClass.value} style={style.value}></div>
+      const loaderList = Array.from({ length: props.repeat }).map(() => loader)
+
+      return <div class={prefixCls}>{props.loading ? loaderList : slots.default?.()}</div>
```
Comment (danranVm) [inline packages/components/skeleton/src/Skeleton.tsx:51]:
```suggestion
      if(!props.loading) {
        return slots.default?.()
      }
      
      const prefixCls = mergedPrefixCls.value
      const loader = <div class={loaderClass.value} style={style.value}></div>
      const loaderList = Array.from({ length: props.repeat }).map(() => loader)

      return <div class={prefixCls} {...attrs}>{loaderList}</div>
```

这样就不会多一层 div

File: packages/components/skeleton/src/Skeleton.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import { computed, defineComponent, normalizeClass } from 'vue'
+
+import { isNumber } from 'lodash-es'
+
+import { useGlobalConfig } from '@idux/components/config'
+
+import { skeletonProps } from './types'
+
+export default defineComponent({
+  name: 'IxSkeleton',
```
Comment (danranVm) [inline packages/components/skeleton/src/Skeleton.tsx:17]:
```suggestion
  name: 'IxSkeleton',
  inheritAttrs: false,
```
为了避免 attrs 被透传到默认插槽，所以这里要设置一下

File: packages/components/skeleton/src/Skeleton.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  props: skeletonProps,
+  setup(props, { slots }) {
```
Comment (danranVm) [inline packages/components/skeleton/src/Skeleton.tsx:19]:
```suggestion
  setup(props, { attrs, slots }) {
```

File: packages/components/skeleton/style/index.less [hunk 9]
```diff
@@ -0,0 +1,35 @@
+@import './themes/default.less';
+@import './mixin.less';
+
+@skeleton-prefix: ~'@{idux-prefix}-skeleton';
```
Comment (danranVm) [inline packages/components/skeleton/style/index.less:4]: 这行可以移除掉，在 prifix.less 里已经自动声明了。

**Review by danranVm — CHANGES_REQUESTED** [review]

File: packages/components/skeleton/demo/Loading.md (hunk 1 above)
Comment (brenner8023) [reply packages/components/skeleton/demo/Loading.md:9]: 我的想法是，有额外生成一个 div，如果给骨架屏设置样式间距，样式间距对插槽内容也同样生效。

File: packages/components/skeleton/docs/Index.zh.md (hunk 2 above)
Comment (brenner8023) [reply packages/components/skeleton/docs/Index.zh.md:27]: fix

File: packages/components/skeleton/docs/Index.zh.md (hunk 3 above)
Comment (brenner8023) [reply packages/components/skeleton/docs/Index.zh.md]: 已经移除了

File: packages/components/skeleton/src/Skeleton.tsx (hunk 4 above)
Comment (brenner8023) [reply packages/components/skeleton/src/Skeleton.tsx]: 已修改

File: packages/components/skeleton/src/Skeleton.tsx (hunk 5 above)
Comment (brenner8023) [reply packages/components/skeleton/src/Skeleton.tsx]: 已修改

File: packages/components/skeleton/src/Skeleton.tsx (hunk 6 above)
Comment (brenner8023) [reply packages/components/skeleton/src/Skeleton.tsx]: 按修改建议改了

File: packages/components/skeleton/src/Skeleton.tsx (hunk 7 above)
Comment (brenner8023) [reply packages/components/skeleton/src/Skeleton.tsx:16]: 已加上

File: packages/components/skeleton/src/Skeleton.tsx (hunk 8 above)
Comment (brenner8023) [reply packages/components/skeleton/src/Skeleton.tsx]: 已加上

**Review by danranVm — APPROVED** [review]

**Pull request merged**